openmc/filter.py

```diff
@@ -25,9 +26,37 @@
 
 class FilterMeta(ABCMeta):
     def __new__(cls, name, bases, namespace, **kwargs):
+        # Check the class name.
         if not name.endswith('Filter'):
             raise ValueError("All filter class names must end with 'Filter'")
+
+        # Create a 'short_name' attribute that removes the 'Filter' suffix.
         namespace['short_name'] = name[:-6]
+
+        # Subclass methods can sort of inherit the docstring of parent class
+        # methods.  If a function is defined without a docstring, most (all?)
+        # Python interpreters will search through the parent classes to see if
+        # there is a docstring for a function with the same name, and they will
+        # use that docstring.  However, Sphinx does not have that functionality.
+        # This chunk of code handles this docstring inheritance manually so that
+        # the autodocumentation will pick it up.
+        if name != 'Filter':
+            # Look for newly-defined functions that were also in Filter.
+            for func_name in namespace:
+                if func_name in Filter.__dict__:
+                    # Inherit the docstring from Filter if not defined.
+                    if isinstance(namespace[func_name], classmethod):
```

Comment:
To really be comprehensive, you should probably have `isinstance(..., (classmethod, staticmethod))` (Thanks to @sethrj for the suggestion).